examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2077 @@
+        # to goal distance.
+        if (
+            action_name != "GRAB_RELEASE"
+            and self._prev_measure["gripped_object_id"] != -1
+        ):
+            object_to_goal_dist_reward = self.get_object_to_goal_dist_reward()
+
+        if (
+            self._episode_success(observations)
+            and self._prev_measure["gripped_object_id"] == -1
+            and action_name == "STOP"
+        ):
+            episode_success_reward = self._rl_config.SUCCESS_REWARD
+
+        reward += (
+            agent_to_object_dist_reward
+            + object_to_goal_dist_reward
+            + gripped_success_reward
+            + episode_success_reward
+        )
+
+        return reward
+
+    def get_agent_to_object_dist_reward(self, observations):
+        """
+            Encourage the agent to move towards the closest object which is not already in place.
+        """
+        curr_metric = self._env.get_metrics()["agent_to_object_distance"]
+        prev_metric = self._prev_measure["agent_to_object_distance"]
+        dist_reward = prev_metric - curr_metric
+
+        self._prev_measure["agent_to_object_distance"] = curr_metric
+
+        return dist_reward
+
+    def get_object_to_goal_dist_reward(self):
+        curr_metric = self._env.get_metrics()["object_to_goal_distance"]
+        prev_metric = self._prev_measure["object_to_goal_distance"]
+        dist_reward = prev_metric - curr_metric
+
+        self._prev_measure["object_to_goal_distance"] = curr_metric
+
+        return dist_reward
+
+    def _episode_success(self, observations):
+        r"""Returns True if object is within distance threshold of the goal.
+        """
+        dist = self._env.get_metrics()["object_to_goal_distance"]
+        if (
+            abs(dist) > self._success_distance
+            or observations["gripped_object_id"] != -1
+        ):
+            return False
+        return True
+
+    def _gripped_success(self, observations):
+        if (
+            observations["gripped_object_id"] >= 0
+            and observations["gripped_object_id"]
+            != self._prev_measure["gripped_object_id"]
+        ):
+            return True
+
+        return False
+
+    def get_done(self, observations):
+        done = False
+        action_name = self._env.task.get_action_name(
+            self._previous_action["action"]
+        )
+        if self._env.episode_over or (
+            self._episode_success(observations)
+            and self._prev_measure["gripped_object_id"] == -1
+            and action_name == "STOP"
+        ):
+            done = True
+        return done
+
+    def get_info(self, observations):
+        info = self.habitat_env.get_metrics()
+        info["episode_success"] = self._episode_success(observations)
+        return info
+
+
+# %%
+import os
+import time
+from collections import defaultdict, deque
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import torch
+import tqdm
```

Comment:
You don't seem to actually use this import anywhere? And if you did, you should consider importing tqdm.auto instead as that would select the write progressbar depending on if its python or ipython